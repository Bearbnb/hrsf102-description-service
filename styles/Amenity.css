/* .amenityHr {
  width: 450px;
  border: 0 none;
  height: 1px;
  color: #eee;
} */

.border {
  border-bottom: 1px solid #EBEBEB;
}

.amenity {
  color: #484848;
  width: 90%;
  margin: 0 auto 0 auto;
}