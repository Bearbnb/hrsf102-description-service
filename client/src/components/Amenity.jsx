import React from 'react';
import PropTypes from 'prop-types';

import styles from '../../../styles/Amenity.css';

const Amenity = ({ availableAmenity }) => {
  const amenityFormatMap = {
    basics: 'Basics',
    guestAccess: 'Guest Access',
    bedAndBath: 'Bed and Bath',
    safetyFeatures: 'Safety Features',
    notIncluded: 'Not Included',
  };

  return (
    <div className={styles.amenity}>
      <h3 className={styles.title}>
        {amenityFormatMap[availableAmenity.amenity]}
      </h3>
      <div className={styles.border} />
      {availableAmenity.amenities.map((amenity => (
        <p>{amenity.name}</p>
      )))}
    </div>
  );
};

Amenity.propTypes = {
  availableAmenity: PropTypes.shape({
    name: PropTypes.string,
    details: PropTypes.string,
  }),
};

Amenity.defaultProps = {
  availableAmenity: {
    amenity: 'basics',
    amenities: [{ name: 'Heating', details: '' }],
  },
};

export default Amenity;
