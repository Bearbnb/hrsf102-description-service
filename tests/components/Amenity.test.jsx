import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Amenity from '../../client/src/components/Amenity';

describe('<Amenity />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Amenity />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
