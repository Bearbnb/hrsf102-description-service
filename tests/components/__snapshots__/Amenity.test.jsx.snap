// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Amenity /> should render properly 1`] = `
<div
  className="amenity"
>
  <h3
    className="title"
  >
    Basics
  </h3>
  <div
    className="border"
  />
  <p>
    Heating
  </p>
</div>
`;
